stl/inc/utility

```diff
@@ -466,6 +459,14 @@ struct pair { // store a pair of values
 
     _Ty1 first; // the first stored value
     _Ty2 second; // the second stored value
+
+private:
+    template <class _Tuple1, class _Tuple2, size_t... _Indices1, size_t... _Indices2>
+    constexpr pair(_Tuple1& _Val1, _Tuple2& _Val2, index_sequence<_Indices1...>, index_sequence<_Indices2...>)
+        : first(_STD _Tuple_get<_Indices1>(_STD move(_Val1))...),
+          second(_STD _Tuple_get<_Indices2>(_STD move(_Val2))...) {}
+
+    pair& operator=(const volatile pair&) = delete;

```

Comment:
Could we please put this back with the other `operator=` overloads? I'm not sure what we're being consistent with, but I believe we generally prefer to keep overload sets together absent other overriding concerns.

Reply:
Reverted. The "consistency" was because of that this overload is non-standard, perhaps only used for suppressing implicit declarations.

Reply:
I vaguely recall that we had to add this overload because MSVC was poorly handling the Standard's mandated explosion of weird overloads here. I strongly agree with Casey here.

It might be time to investigate whether we can remove any occurrences of this non-Standard "delete `op=` for cv" pattern, but that should be a separate PR if so.

Reply:
Wow do I wish we had commented this hack.

Reply:
> It might be time to investigate whether we can remove any occurrences of this non-Standard "delete `op=` for cv" pattern, but that should be a separate PR if so.

Currently this pattern exists for `pair` and `tuple`.

These overloads are copy assignment operators. I guess we can't directly remove them, since [[class.copy.assign]/2](https://eel.is/c++draft/class.copy.assign#2) would require a copy assignment operator taking `const T&` to be implicitly declared and deleted.